### PR TITLE
KER-195 Registered user can comment anonymously

### PR DESCRIPTION
**Pull request by mikkojamG**
# KER-195 Registered user can comment anonymously

- Registered user (not admin) can optionally write their comment anonymously
- Refactoring BaseCommentForm into function component

Screenshot 2023-08-08 at 12 17 40

### CODE REVIEW

File: src/components/BaseCommentForm.js [hunk 1]
```diff
@@ -45,273 +51,332 @@ Leaflet.Marker.prototype.options.icon = new Leaflet.Icon({
...
 
-  handleErrorStates(errors) {
-    this.setState({
-      commentRequiredError: errors.includes('commentRequiredError'),
-      commentOrAnswerRequiredError: errors.includes('commentOrAnswerRequiredError'),
-      imageTooBig: errors.includes('imageTooBig')
+  const isImageTooBig = (images) => {
+    let imageTooBig = false;
+    Array.from(images).forEach((image) => {
+      if (image.size > 1000000) {
```
Comment (voneiden) [inline src/components/BaseCommentForm.js:300]: Optional opportunity to refactor this to be an environment variable from config.js

File: src/components/BaseCommentForm.js [hunk 2]
```diff
@@ -320,347 +385,382 @@ export class BaseCommentForm extends React.Component {
...
+            </div>
+          ) : (
+            formData.images.map((image, key) => (
+              <img
+                style={{ marginRight: 10 }}
+                alt=""
+                src={image.image}
+                width={image.width < 100 ? image.width : 100}
+                height={image.height < 100 ? image.width : 100}
+                key={key + Math.random()} //eslint-disable-line
```
Comment (voneiden) [inline src/components/BaseCommentForm.js:714]: OK this is old code, but how about using image.image (base64 url string) + index number as key? Not foolproof.. alternatively generate random uuids when the images are processed in handleChange ?

**Review by voneiden — APPROVED** [review]
LGTM, left few optional comments.

File: src/components/BaseCommentForm.js [hunk 3]
```diff
@@ -45,273 +51,332 @@ Leaflet.Marker.prototype.options.icon = new Leaflet.Icon({
...
+      nickname,
+      commentText,
+      geojson,
+      images,
+      pinned,
+      mapCommentText,
+      imageTooBig,
+    } = formData;
+
+    let setNickname = nickname === '' ? nicknamePlaceholder : nickname;
```
[the remark this answers is not in the document]
Comment (mikkojamG) [reply src/components/BaseCommentForm.js]: https://github.com/City-of-Helsinki/kerrokantasi-ui/pull/958/commits/fe0e02c46761f286c0a3c2455f5dc45ba13c279a

File: src/components/BaseCommentForm.js (hunk 1 above)
Comment (mikkojamG) [reply src/components/BaseCommentForm.js]: https://github.com/City-of-Helsinki/kerrokantasi-ui/pull/958/commits/fe0e02c46761f286c0a3c2455f5dc45ba13c279a

File: src/components/BaseCommentForm.js (hunk 2 above)
Comment (mikkojamG) [reply src/components/BaseCommentForm.js]: https://github.com/City-of-Helsinki/kerrokantasi-ui/pull/958/commits/fe0e02c46761f286c0a3c2455f5dc45ba13c279a

File: src/components/BaseCommentForm.js (hunk 1 above)
Comment (voneiden) [reply src/components/BaseCommentForm.js]: There's also https://github.com/City-of-Helsinki/kerrokantasi-ui/blob/master/src/config.js but upon further inspection I have no idea if and how it is populated

**Review by voneiden — APPROVED** [review]

**Pull request merged**